src/librustc_trans/back/write.rs

```diff
@@ -557,10 +565,13 @@ unsafe fn optimize_and_codegen(cgcx: &CodegenContext,
     // machine code, instead copy the .o file from the .bc
     let write_bc = config.emit_bc || config.obj_is_bitcode;
     let rm_bc = !config.emit_bc && config.obj_is_bitcode;
+    let write_asm = config.emit_asm || config.obj_needs_as;
```

Comment:
I think if we call `write::run_assembler` or hook into `no_integrated_as` these changes may not be necessary. 

That being said, I'm not sure whether the code path with `no_integrated_as` still works, I can't imagine anyone's used that in forever...